### PR TITLE
Use Ruby 2.2.10

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Part 1 of #2891 

Bumps Ruby version to `2.2.10`, giving many (over 1 thousand?) improvements:

- 2.2.0 https://svn.ruby-lang.org/repos/ruby/tags/v2_2_0/ChangeLog
- 2.2.1 https://svn.ruby-lang.org/repos/ruby/tags/v2_2_1/ChangeLog
- 2.2.2 https://svn.ruby-lang.org/repos/ruby/tags/v2_2_2/ChangeLog
- 2.2.3 https://svn.ruby-lang.org/repos/ruby/tags/v2_2_3/ChangeLog
- 2.2.4 https://svn.ruby-lang.org/repos/ruby/tags/v2_2_4/ChangeLog
- 2.2.5 https://svn.ruby-lang.org/repos/ruby/tags/v2_2_5/ChangeLog
- 2.2.6 https://svn.ruby-lang.org/repos/ruby/tags/v2_2_6/ChangeLog
- 2.2.7 https://svn.ruby-lang.org/repos/ruby/tags/v2_2_7/ChangeLog
- 2.2.8 https://svn.ruby-lang.org/repos/ruby/tags/v2_2_8/ChangeLog
- 2.2.9 https://svn.ruby-lang.org/repos/ruby/tags/v2_2_9/ChangeLog
- 2.2.10 https://svn.ruby-lang.org/repos/ruby/tags/v2_2_10/ChangeLog
#### What should we test?

- [x] Provision a staging server with: https://github.com/openfoodfoundation/ofn-install/pull/530
- [x] Stage this PR
- [x] Check everything looks OK
- [x] Try running the CI build with `2.2.10`

#### Release notes

Bumped Ruby version from `2.1.9` to `2.2.10`

Changelog Category: Changed

### CODE REVIEW

**Comment by Matt-Yorkley** [comment]
This is now staged on UK, and seems to be working with `2.2.10` :tada:

**Comment by Matt-Yorkley** [comment]
2 build errors: https://semaphoreci.com/openfoodfoundation/openfoodnetwork-2/branches/pull-request-4301/builds/6

Looks like they are both from a display issue on the table for the `xero_invoices` report.

**Comment by luisramos0** [comment]
so this needs to go back to "in dev" until the build errors are solved, correct?

**Comment by Matt-Yorkley** [comment]
@mkllnk you mentioned you did this for Ceres and needed to patch a similar spec to make it work? Any clues here?

**Comment by mkllnk** [comment]
The spec that failed in Fair Food was just the order of an unsorted array. These failures look different. We have too look into the code.

**Comment by Matt-Yorkley** [comment]
These 2 small build errors need looking at and fixing before this can move forward. Moving to Tech Debt backlog if anyone wants to take a look.

**Comment by luisramos0** [comment]
I want to give it a try.

**Comment by luisramos0** [comment]
I have created #4374 to represent the work to fix the specs.

**Comment by luisramos0** [comment]
green build. one liner...
I think there were some changes in the way the params get to the controller. The keys are strings and need to be converted to symbols so that the code afterwards can access the values using symbols as keys.
It's a bit intriguing this only breaks this controller...

back to code review 💪

**Comment by Matt-Yorkley** [comment]
Awesome! Yeah, it's pretty weird that this issue only affected one controller..?

Github says I'm not allowed to approve this PR, but I approve!

**Comment by sigmundpetersen** [comment]
so this closes #4374 @luisramos0?

**Comment by luisramos0** [comment]
yes @sigmundpetersen thanks!

File: Gemfile [hunk 1]
```diff
@@ -15,7 +15,8 @@ gem 'nokogiri', '>= 1.6.7.1'
 gem "order_management", path: "./engines/order_management"
 gem 'web', path: './engines/web'
 
-gem 'pg'
+gem 'activerecord-postgresql-adapter'
```
Comment (sauloperez) [inline Gemfile]: This looks a bit odd to me. We have another Rails 4.2 and Ruby 2.3.0 app and we don't need this gem. Was it due to the upgrade of `pg`?

Checking the code of the gem at https://github.com/larskanis/activerecord-postgresql-adapter I don't understand it either.

Comment (Matt-Yorkley) [reply Gemfile]: I think bundler failed when I ran it, and the output suggested we needed this.

Comment (Matt-Yorkley) [reply Gemfile]: I think it adds compatibility for Rails 3 activerecord gems. Some of them try to include older versions of pg directly, and break.

Comment (Matt-Yorkley) [reply Gemfile]: You can see this if you try to run the app without the adapter gem.

**Comment by mkllnk** [comment]
@luisramos0 I ran this branch three times on my Semaphore account with Ruby 2.2.10 but I get failing specs that pass locally:

- https://semaphoreci.com/mkllnk/openfoodnetwork/branches/ruby-2-2-10/builds/1
- https://semaphoreci.com/mkllnk/openfoodnetwork/branches/ruby-2-2-10/builds/2
- https://semaphoreci.com/mkllnk/openfoodnetwork/branches/ruby-2-2-10/builds/3

**Comment by luisramos0** [comment]
@mkllnk the specs failing are all known to me, I mean, they failed at some point... I wonder if they have some sort of flakyness to them... anyway, I think we should go ahead and test this.

**Comment by Matt-Yorkley** [comment]
> I ran this branch three times on my Semaphore account with Ruby 2.2.10 but I get failing specs that pass locally

It looks like your CI environment is running in Spanish, but the those tests are looking for English strings?

```
Failure/Error: expect(rendered).to have_content("Print Invoices")
       expected to find text "Print Invoices" in "{{ 'spree.admin.orders.index.results_found' | t:{number: pagination.results} }} {{ 'spree.admin.orders.index.viewing' | t:{start: ((pagination.page -1) * pagination.per_page) +1, end: ((pagination.page -1) * pagination.per_page) + orders.length} }} Imprimir Faturas Distribuidor Concluído Em ▲ ▼ Número ▲ ▼ Estado ▲ ▼ Estado do Pagamento ▲ ▼ Estado do envio ▲ ▼ Email do/a Consumidor/a ▲ ▼ Total ▲ ▼ {{order.distributor_name}} {{order.completed_at}} {{order.number}} Nota {{'js.admin.orders.order_state.' + order.state | t}} {{'js.admin.orders.payment_states.' + order.payment_state | t}} {{'js.admin.orders.shipment_states.' + order.shipment_state | t}} {{order.email}} A carregar « Primeiro Anterior … {{$index+1}} … Seguinte Último » Nenhuma encomenda encontrada"
```

**Comment by luisramos0** [comment]
ah, probably same problem I had on another PR, probably fixed by this commit:
https://github.com/openfoodfoundation/openfoodnetwork/pull/4057/commits/f45eb35eb1acf9f8531175f0a0050046f819884e

**Comment by Matt-Yorkley** [comment]
**French Staging** has now been provisioned with https://github.com/openfoodfoundation/ofn-install/pull/530 and can now be deployed with this PR and tested.

**Comment by RachL** [comment]
@Matt-Yorkley what type of test do we need to run?

**Comment by Matt-Yorkley** [comment]
It's a bit like the last one in #4250. Something like a release test? The build is green so I guess we can be fairly confident, but it's a big change...

**Comment by Matt-Yorkley** [comment]
I might not be online much for the next week, so just a note if this gets merged: we'll need to change the Semaphore build settings to use `2.2.10` and provision all servers (including staging) with https://github.com/openfoodfoundation/ofn-install/pull/530 as soon as this is in `master`.

**Comment by luisramos0** [comment]
provision all servers (including staging) can be done at any time right? it wont change the ruby version of the app, just install the version, correct?

**Comment by Matt-Yorkley** [comment]
Yes, it should be fine to do it any time.

**Comment by RachL** [comment]
Ok so I've taken the same testing approach than #4250: I did a sanity check:

1- Login
2- Create a product and it appears correctly in the shop
3- Checkout (stripe & cash)
4- See the order on the order view (+ capture and ship and edit)

The rest will be tested during release testing like last time. So I'm moving this to ready to go, if you think in needs deeper testing, move it back :-)

**Comment by RachL** [comment]
Forgive me if I'm wrong but last time I think updating the docker was forgotten and added later, maybe something to do at the same time as provisioning servers? https://github.com/openfoodfoundation/openfoodnetwork/pull/4350

**Comment by Matt-Yorkley** [comment]
I think it needs a separate issue where we have Docker fetch the version from the codebase instead of having to change that number every time.

**Comment by sauloperez** [comment]
> I think it needs a separate issue where we have Docker fetch the version from the codebase instead of having to change that number every time.

totally, that's what I thought doing last time.

**Comment by luisramos0** [comment]
adding the blocked lavel. I think I'll have time to take care of this tomorrow:
- provision all servers with ofn-install#530
- change semaphore to use 2.2.10
- merge this PR to master
- after this all PR builds will fail until they are rebased to get 2.2.10

@sauloperez will you have time to prepare a release tomorrow? maybe I'll start the day doing this and you prepare the release afterwards?

**Comment by sauloperez** [comment]
I was right now looking into it to take it on but this would mean removing things from my todo list. I'd much appreciate if you could do exactly that while I deal with the regular release.

**Pull request merged**